### PR TITLE
wolves, and point handling

### PR DESCRIPTION
**Pull request by Skullyton**
Wolves

FAR FROM DONE, even still has some hitlers dropped in from testing

TODO:
- [x] Pack mentality,
- [x]  - Stay in proximity of the pack alpha unless ordered otherwise
- [x]  - Accept orders from pack alpha
- [x] Hunger level, where they'll start hunting if they're too hungry 
- [x]  Can feed them like a carp baby to try and tame them, which would otherwise challenge the alpha
- [x]  Be able to point at the wolves and around the area to order loyal wolves to do things (attack things)

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/wolf.dm [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+					to_chat(world, "[src] is challenging [alpha_challenger]")
+					challenge(alpha_challenger)
+
+/mob/living/simple_animal/hostile/wolf/CanAttack(var/atom/the_target)
+	//WE DON'T ATTACK INVULNERABLE MOBS (such as etheral jaunting mobs, or passengers of the adminbus)
+	if(the_target.flags & INVULNERABLE)
+		return 0
+	if(istype(the_target, /mob/dead/observer))
+		return 0 //Stop eating ghosts!
+	if(istype(the_target, alpha_target) || istype(the_target, alpha_challenger))//Who the alpha has specified
```
Comment (Skullyton) [inline code/modules/mob/living/simple_animal/hostile/wolf.dm]: Good point, this makes a whole lotta sense come to think of it

**Comment by DrCelt** [comment]
🐺🐺🐺🐺

File: code/modules/mob/living/simple_animal/hostile/wolf.dm [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+#define HUNGRY 2
+#define VHUNGRY 1
+#define STARVING 0
+
+#define MOVECOST 0.5
+#define STANDCOST 0.5
+#define REGENCOST 20
+
+#define MAXALPHADIST 7
+/* TODO: Pack mentality - Wolves will generally stick around the 'alpha', at least within 6 tiles, unless hunting
```
Comment (DrCelt) [inline code/modules/mob/living/simple_animal/hostile/wolf.dm]: TODOne?

File: code/modules/mob/living/simple_animal/hostile/wolf.dm [hunk 3]
```diff
@@ -253,24 +290,110 @@
...
+
+			stance = HOSTILE_STANCE_ATTACK
+			alpha_stance = ALPHAATTACK
+			alpha_target = target
+		if(istype (target, /turf)) //We go!
+			alpha_stance = ALPHAMOVE
+			alpha_target = target
+
+/mob/living/simple_animal/hostile/wolf/pointed_at(mob/pointer)
+	if(pointer == point_last)
```
Comment (Skullyton) [inline code/modules/mob/living/simple_animal/hostile/wolf.dm]: Well this is a bit of an oversight. Why is this even here?

**Comment by Skullyton** [comment]
There is also currently nothing stopping nutrition from going into the negative. Only downside to this is you have a very angry wolf who'll take a lot of batmeat sliders to cheer up and stop mauling the clown

File: code/modules/mob/living/simple_animal/hostile/wolf.dm [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+			if(STARVING)
+				to_chat(user, "<span class='warning'>It looks starving!</span>")
+		if(pack_alpha == user)
+			to_chat(user, "<span class='info'>It seems friendly to you.</span>")
+		var/remaining_health_percent = round(((health/maxHealth)*100))
+		switch(remaining_health_percent)
+			if(100 to 60)
+				//Do nuthin
+			if(59 to 30)
+				to_chat(user, "<span class='warning'>It seems quite hurt.</span>")
```
Comment (DrCelt) [inline code/modules/mob/living/simple_animal/hostile/wolf.dm]: The yanks have a different subtle definition for quite than we do

**Comment by DrCelt** [comment]
Ready?

**Pull request merged**